src/plugins/intel_gpu/src/plugin/ops/gather.cpp

```diff
@@ -46,6 +48,43 @@ void CreateGatherOpBase(Program& p, const std::shared_ptr<T>& op, const int64_t
     // Dynamic path will do shape infer internally, so no need to pass valid out shape for that case
     ov::Shape out_shape = op->get_output_partial_shape(0).is_static() ? op->get_output_shape(0) : ov::Shape{};
 
+    // Update output_shape in case of scalar indice
+    bool need_reshape = false;
+    auto out_shape_original = out_shape;
+    if (!p.use_new_shape_infer() && op->get_output_partial_shape(0).is_static()) {
+        auto input1_shape = op->get_input_shape(1);
+        if (input1_shape.size() == 0 && batch_dim == 0) {
+            need_reshape = true;
+
+            auto new_axis = axis;
+            if (new_axis < 0) {
+                new_axis += op->get_input_shape(0).size();
+            }
+            out_shape.push_back(1);
+            for (int i = out_shape.size() - 1; i > new_axis ; i--) {
+                out_shape[i] = out_shape[i-1];
+            }
+            out_shape[new_axis] = 1;
+        }
+    }
+
+    // WA for NMS->Gather construction. NMS fills part of the output blob by the -1 if these values
+    // must not be taken into account.
+    // CPU also uses this like of WA.
+    if (support_neg_ind) {
+        const auto& rti = op->get_rt_info();
+        const auto& reverse = rti.find("dontReverseIndices");
+        if (reverse != rti.end()) {
+            support_neg_ind = false;
+        }
```

Comment:
BTW, currently its like :
if support_neg_ind = false 
    then the -1 is to be used as is in the gather_ref.cl kernel. 

Then won't there be an error with accessing to indices[-1]  ? 
Shouldn't we do like, if support_neg_ind = false and the indices is -1, just return without doing anything?